Source/WebKit/Platform/IPC/Connection.cpp

```diff
@@ -529,6 +532,11 @@ Error Connection::flushSentMessages(Timeout timeout)
 void Connection::invalidate()
 {
     m_isValid = false;
+
+#if ENABLE(IPC_TESTING_API)
+    setErrorString({ });
```

Comment:
I think it might be useful to keep the error state if invalidate is called, and no downside as far as I see.
An error handling codepath might run common code that runs invalidate(), which would then lose the error information that might be accessed by a test that uses the feature to test the error condition.